### DISCUSSION TITLE
TP3 - Duda de cadenas

### DISCUSSION THREAD

**Discussion by rinrot** [INGCOM-UNRN-P1/cursada-2023#99] [Preguntas y respuestas] [answered]
Hola! estoy haciendo el ejercicio 2.1 del TP3 (Implementar una función que cuente el largo de una cadena de texto de manera segura, La función debe retornar el largo de la cadena o el código de error correspondiente.) y queria saber si es necesario retornar si la cadena overflowea o si es un puntero nulo, porque si puedo detectar eso fuera de la funcion `largo_seguro` me puedo ahorrar el tener que detectarlo cada vez que tenga que utilizar otra funcion.

**Comment by martinvilu** [comment]
¿Y como sería eso de fuera de la función?

**Reply by rinrot** [reply]
ya lo borre, pero basicamente la idea era que detecte si es un valor nulo o si es overflow antes de entrar a las otras funciones de esta manera:

```
#include <stdio.h>

#define CARACTERES 100
#define VACIO '\0'

main()
{
    char cadena[CARACTERES];
    int output;
    int opcion=0;
    printf("Escriba algo aqui: ");
    fgets(cadena, sizeof(cadena), stdin);

    if(overflow) ----> todavia no se como se hace este 
    {
        printf("CAPACIDAD DE CARACTERES SUPERADA\n");
    }
    else if(cadena[0]==VACIO)
    {
        printf("puntero nulo\n");
    }
    else
    {
        printf("por favor, elija la funcion...
     }
}
```

**Reply by martinvilu** [reply, marked as answer]
La idea de tener las consideraciones de "seguro" es para que la función trabaje incluso cuando no hay suficiente espacio, o el `\0` se perdió por alguna razón.

Algunas de las funciones más nuevas de la librería de C que trabajan con cadenas, trabajan con este principio para evitar problemas.
